release/platform/platform.go

```diff
@@ -119,7 +119,7 @@ func DetectLocal() (Platform, error) {
 		}
 		return pf, nil
 	case "Darwin":
-		pf, ok := GetByVariant("macos")
+		pf, ok := GetByVariant("macos-arm")
```

Comment:
With the caveat that I don't totally understand the build script: is this sufficient? (I don't _think_ so, but I'm not totally sure.) Presumably we still want to support building on x86 macs, and I don't think this would let that happen. (I think also this interacts with the other change I suggested, to use the name/arch pair.)